### DISCUSSION TITLE
How to change property from ref?

### DISCUSSION THREAD

**Discussion by VicoErv** [Tresjs/tres#863] [Q&A] [answered]
I try to change
```
const splineMesh = meshRef.current;
const position = splineMesh.geometry.attributes.position;
position.setXYZ(i, point.x, point.y, point.z);
position.needsUpdate = true;
```

but I got warning from reactivity
`[Vue warn] Set operation on key "position" failed: target is readonly. `

how to properly change the properties?

**Comment by andretchen0** [comment, marked as answer]
I'm going to close here and point you to the Tres Discord:

https://discord.com/invite/WYj3pjpPUK

It's a lot faster to get support over there.

To help people help you, please post a whole minimal example that produces the bug.

**Reply by VicoErv** [reply]
thank you, I just not understand. But now I get it, that I should use `shallowRef`, thanks for pointing out the discord invitation
